### PR TITLE
Add a new parallelCoordinate chart

### PR DESCRIPTION
**Pull request by mgalloue**
##### ParallelCoordinate refactoring
 - Support persistence on brush and sorting even if dimensions are disabled, added or removed or data are updated.
 - Rename dimensionNames to dimensionData (array of object containing key, format and tooltip)
 -  Support custom color and stroke for each data

##### New ParallelCoordinateChart
- Support Tooltip on axis and on lines
- Support legend and disabled dimensions
- Reset brush and reset sorting button 
- Support interactive update on data and dimensions

### CODE REVIEW

File: src/models/parallelCoordinatesChart.js [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+                            resetBrushButton.style("visibility", "hidden");
+                        });
+
+                        resetBrushButton.node().parentNode.insertBefore(container.node(), resetBrushButton.node().nextSibling);
+                    }
+
+                    if (resetSortingButton === null) {
+                        resetSortingButton = d3.select(container.node().parentNode).append("button")
+                            .attr("class", "nv-resetSortingButton")
+                            .text("Reset Sorting");
```
Comment (liquidpele) [inline src/models/parallelCoordinatesChart.js]: Why are we including button controls in the actual code?    This should probably be put in the example HTML and then affects the chart either through setting changes or dispatch.

File: examples/documentation.html [hunk 2]
```diff
@@ -767,18 +768,35 @@ <h3 class="option"><span class="expand all">[ <a href="javascript:$('#optionlist
...
-            examples: [["axis1", "axis2", "axis3"]]
+            desc: "Deprecated. Use dimensionData instead. "
         },
         dimensionFormats: {
-            desc: "D3 format for each x axis",
-            default: "Use the default formatter scale.tickFormat",
-            examples: [["%", "", "0.2f"]]
+            desc: "Deprecated. Use dimensionData instead. "
+        },
+		dimensionData: {
```
Comment (liquidpele) [inline examples/documentation.html]: Your indention is off here

File: examples/documentation.html [hunk 3]
```diff
@@ -767,18 +768,35 @@ <h3 class="option"><span class="expand all">[ <a href="javascript:$('#optionlist
...
+            desc: "Deprecated. Use dimensionData instead. ",
+            default: "[]",
+            examples: [[
+				{key:"P1",format: "0.2f", tooltip: "Parameter 1"},
+				{key:"P2",format: "%", tooltip: "Parameter 2"}]
+			]
+        },
+		displayBrush: {
```
Comment (liquidpele) [inline examples/documentation.html]: indention

File: examples/documentation.html [hunk 4]
```diff
@@ -767,18 +768,35 @@ <h3 class="option"><span class="expand all">[ <a href="javascript:$('#optionlist
...
+            desc: "Support brush on axis",
+            default: true,
+            examples: [false]
         },
+		filters: {
```
Comment (liquidpele) [inline examples/documentation.html]: indention

File: src/models/parallelCoordinatesChart.js [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+        chart.options = nv.utils.optionsFunc.bind(chart);
+
+        chart._options = Object.create({}, {
+            // simple options, just get/set the necessary values
+            width: { get: function () { return width; }, set: function (_) { width = _; } },
+            height: { get: function () { return height; }, set: function (_) { height = _; } },
+            showLegend: { get: function () { return showLegend; }, set: function (_) { showLegend = _; } },
+            showControls: { get: function () { return showControls; }, set: function (_) { showControls = _; } },
+            defaultState: { get: function () { return defaultState; }, set: function (_) { defaultState = _; } },
+            dimensionData: { get: function () { return dimensionData; }, set: function (_) { dimensionData = _; } },
```
Comment (liquidpele) [inline src/models/parallelCoordinatesChart.js]: You removed dimentionFormats... you should keep it but add a "this is deprecated" message like we did with several other options... also include a comment about what version it was deprecated after so we know when we can remove it completely.

File: src/models/parallelCoordinates.js [hunk 6]
```diff
@@ -89,7 +144,7 @@ nv.models.parallelCoordinates = function() {
             //Add missing value line at the bottom of the chart
             var missingValuesline, missingValueslineText;
             var step = x.range()[1] - x.range()[0];
-            var axisWithMissingValues = [];
+            if (!isNaN(step)) {
```
Comment (liquidpele) [inline src/models/parallelCoordinates.js]: You made this an if block but didn't indent the interior...

File: src/models/parallelCoordinates.js [hunk 7]
```diff
@@ -183,55 +286,59 @@ nv.models.parallelCoordinates = function() {
...
+                            return [x(p.key), y[p.key](min)];
                         }
-
-                        return [x(p), y[p](min)];
                     }
 
                     //If parallelCoordinate contain missing values show the missing values line otherwise, hide it.
-                    if(axisWithMissingValues.length > 0) {
+                    if (missingValuesline !== undefined) {
+                        if (axisWithUndefinedValues.length > 0 || displayMissingValuesline) {
```
Comment (liquidpele) [inline src/models/parallelCoordinates.js]: you added an if block but didn't indent the interior.

File: src/models/parallelCoordinatesChart.js (hunk 5 above)
Comment (mgalloue) [inline src/models/parallelCoordinatesChart.js]: I kept it on parallelCoordinates.js and added the "this is deprecated" message.

File: src/models/parallelCoordinatesChart.js (hunk 1 above)
Comment (mgalloue) [inline src/models/parallelCoordinatesChart.js]: @liquidpele sure, I moved it.

**Comment by mgalloue** [comment]
@liquidpele any update on this pull request?

**Comment by liquidpele** [comment]
Can you make this PR to master?   That's the latest code at this point... I really need to just delete the development branch.   I still like the idea of having it, but everyone kept sending PR's to master anyway so it just didn't work out.  

Thanks!

**Comment by mgalloue** [comment]
PR merged on master branch.

**Pull request closed without merging**